lib/PipelineC/PipelineC.cpp

```diff
@@ -915,31 +913,31 @@ bool rp_diff_map_is_empty(rp_diff_map *map) {
 }
 
 rp_error *rp_error_create() {
-  return new rp_error(nullptr);
+  return new rp_error();
 }
 
 bool rp_error_is_success(rp_error *error) {
-  return error->get() == nullptr;
+  return error->index() == 0;
 }
 
 bool rp_error_is_document_error(rp_error *error) {
-  return std::holds_alternative<rp_document_error>(**error);
+  return std::holds_alternative<rp_document_error>(*error);
 }
 
 rp_simple_error *rp_error_get_simple_error(rp_error *error) {
-  if (error->get() == nullptr
-      || not std::holds_alternative<rp_simple_error>(**error))
+  if (error->index() == 0
+      || not std::holds_alternative<rp_simple_error>(*error))
     return nullptr;
 
-  return &std::get<rp_simple_error>(**error);
+  return &std::get<rp_simple_error>(*error);
 }
 
 rp_document_error *rp_error_get_document_error(rp_error *error) {
-  if (error->get() == nullptr
-      || not std::holds_alternative<rp_document_error>(**error))
+  if (error->index() == 0
```

Comment:
Use [`std::holds_alternative`](https://en.cppreference.com/w/cpp/utility/variant/holds_alternative).

Reply:
Re-checked, the test was redundant. Dropped